text/0000-volta-run.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: volta_run
+- Start Date: 2019-09-14
+- RFC PR: (leave this empty)
+- Volta Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a `volta run` command, allowing users to execute tools with an override on the detected platform.
+
+# Motivation
+[motivation]: #motivation
+
+Volta currently will automatically detect the platform (Node runtime and yarn version) based on the context of where you are in the file system. However, there are occasionally times when you want to run a command using a different version of Node or Yarn than what Volta detects, without changing the pinned or default versions. To support those use-cases, we should provide a command that allows users to execute a command with a platform specified on the command-line.
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+We will need to document the existence of `volta run` and how the users can specify the versions of Node and Yarn when running a command. This will be provided on the command line by StructOpt, however it should also be explained in detail in the documentation. We should also document how the overall platform is resolved if the user specifies only one of Node or Yarn, so that the behavior is clear to users.
```

Comment:
```suggestion
We will need to document the existence of `volta run` and how the users can specify the versions of Node and Yarn when running a command. This will be provided on the command line help output, however it should also be explained in detail in the documentation. We should also document how the overall platform is resolved if the user specifies only one of Node or Yarn, so that the behavior is clear to users.
```

`StructOpt` seems like a _somewhat_ unimportant technical detail here, and I think the userland thing that you meant was that the `volta` usage output would include info on `run`.

Reply:
> We should also document how the overall platform is resolved if the user specifies only one of Node or Yarn, so that the behavior is clear to users.

Do you mean in the website? In the help output? Both?

Reply:
Good question! My intention was on the website, where we can be more long-form about the nuances of the command.